packages/client/src/index.ts

```diff
@@ -102,141 +89,145 @@ export class RelayClient extends EventEmitter {
     shopEvent: schema.IShopEvent,
   ): Promise<schema.EventWriteResponse> {
     await this.connect();
+
+    shopEvent.nonce = this.eventNonceCounter++;
+    shopEvent.timestamp = { seconds: Date.now() / 1000 };
     const shopEventBytes = schema.ShopEvent.encode(shopEvent).finish();
     const sig = await this.keyCardWallet.signMessage({
       message: { raw: shopEventBytes },
     });
     const signedEvent = {
-      signature: hexToBytes(sig),
+      signature: { raw: hexToBytes(sig) },
       event: {
         type_url: "type.googleapis.com/market.mass.ShopEvent",
         value: shopEventBytes,
       },
     };
     const eventWriteRequest = {
-      event: signedEvent,
+      eventWriteRequest: {
+        events: [signedEvent],
+      },
     };
-    await this.encodeAndSend(schema.EventWriteRequest, eventWriteRequest);
+    const res = await this.encodeAndSend(eventWriteRequest);
     //Passing current KC address as signer for event verification.
-    this.eventStream.outgoingEnqueue(shopEvent, this.keyCardWallet.address);
+    this.eventStream.outgoingEnqueue(
+      shopEvent,
+      this.keyCardWallet.address,
+      res.requestId,
+    );
+    return res.requestId;
   }
 
   async shopManifest(manifest: schema.IShopManifest, shopId: `0x${string}`) {
-    const id = (manifest.eventId = eventId());
-    manifest.shopTokenId = hexToBytes(shopId);
-    await this.sendShopEvent({
-      shopManifest: manifest,
+    manifest.tokenId = { raw: hexToBytes(shopId) };
+    return await this.sendShopEvent({
```

Comment:
There is no point on await here
```suggestion
    return  this.sendShopEvent({
```